// @nypl/dgx-react-footer/dist/footerConfig.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
var data = {
  nyplLinks: [[{
    name: 'Accessibility',
    link: 'http://www.nypl.org/accessibility'
  }, {
    name: 'Press',
    link: 'http://www.nypl.org/help/about-nypl/media-center'
  }, {
    name: 'Careers',
    link: 'http://www.nypl.org/careers'
  }, {
    name: 'Space Rental',
    link: 'http://www.nypl.org/spacerental'
  }], [{
    name: 'Privacy Policy',
    link: 'http://www.nypl.org/help/about-nypl/legal-notices/privacy-policy'
  }, {
    name: 'Other Policies',
    link: 'http://www.nypl.org/policies'
  }, {
    name: 'Terms & Conditions',
    link: 'http://www.nypl.org/terms-conditions'
  }, {
    name: 'Governance',
    link: 'http://www.nypl.org/help/about-nypl/leadership/board-trustees'
  }], [{
    name: 'Rules & Regulations',
    link: 'http://www.nypl.org/help/about-nypl/legal-notices/rules-and-regulations'
  }, {
    name: 'About NYPL',
    link: 'http://www.nypl.org/help/about-nypl'
  }, {
    name: 'Language',
    link: 'http://www.nypl.org/language'
  }]],
  socialMedia: [{
    name: 'NYPL on Facebook',
    link: 'https://www.facebook.com/nypl',
    className: 'icon-facebook',
    component: 'FaceBookIcon',
    iconId: 'face-book-id-title'
  }, {
    name: 'NYPL on Twitter',
    link: 'https://twitter.com/nypl',
    className: 'icon-twitter',
    component: 'TwitterIcon',
    iconId: 'twitter-id-title'
  }, {
    name: 'NYPL on Instagram',
    link: 'https://instagram.com/nypl',
    className: 'icon-instagram',
    component: 'InstagramIcon',
    iconId: 'instagram-id-title'
  }, {
    name: 'NYPL on Youtube',
    link: 'https://www.youtube.com/user/NewYorkPublicLibrary',
    className: 'icon-youtube',
    component: 'YoutubeIcon',
    iconId: 'youtube-id-title'
  }]
};

exports.default = data;
